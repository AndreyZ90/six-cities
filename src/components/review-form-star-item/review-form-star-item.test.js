import React from 'react';
import TestRenderer from 'react-test-renderer';

import ReviewFormStarItem from '@/components/review-form-star-item/review-form-star-item';

describe(`ReviewFormStarItem component snapshot`, () => {
  test(`Should correctly render ReviewFormStarItem component (default)`, () => {
    const tree = TestRenderer.create(
        <ReviewFormStarItem
          rating={5}
          currentRating={4}
          title={`perfect`}
          onRatingChange={() => {}}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`Should correctly render ReviewFormStarItem component (active)`, () => {
    const tree = TestRenderer.create(
        <ReviewFormStarItem
          rating={5}
          currentRating={5}
          title={`perfect`}
          onRatingChange={() => {}}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
