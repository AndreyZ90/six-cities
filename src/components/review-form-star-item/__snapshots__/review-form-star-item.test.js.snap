// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewFormStarItem component snapshot Should correctly render ReviewFormStarItem component (active) 1`] = `
Array [
  <input
    checked={true}
    className="form__rating-input visually-hidden"
    id="5-star"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={5}
  />,
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="5-star"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height={33}
      width={37}
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>,
]
`;

exports[`ReviewFormStarItem component snapshot Should correctly render ReviewFormStarItem component (default) 1`] = `
Array [
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    id="5-star"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={5}
  />,
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="5-star"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height={33}
      width={37}
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>,
]
`;
